includes/class-wc-gateway-cardknox.php

```diff
@@ -416,8 +416,24 @@ public function get_payment_data($post_data){
             $post_data['xCVV'] = wc_clean( $_POST['xCVV'] );
             $post_data['xExp'] = wc_clean( $_POST['xExp'] );
         }
+        $this->validate_payment_data($post_data);
         return $post_data;
+  }
+
+  public function validate_payment_data($post_data){
+    if ($this->is_unset_or_empty($post_data['xToken'])) {
+      if ($this->is_unset_or_empty($post_data['xCardNum'])) {
+        throw new WC_Data_Exception("wc_gateway_cardknox_process_payment_error", "Invalid card number", 400);
+      }
+      if ($this->is_unset_or_empty($post_data['xCVV'])) {
+        throw new WC_Data_Exception("wc_gateway_cardknox_process_payment_error", "Invalid cvv", 400);
+      }
```

Comment:
```suggestion
        throw new WC_Data_Exception("wc_gateway_cardknox_process_payment_error", "Required: card number", 400);
      }
      if ($this->is_unset_or_empty($post_data['xCVV'])) {
        throw new WC_Data_Exception("wc_gateway_cardknox_process_payment_error", "Required: cvv", 400);
      }
```

Reply:
Hello @ymwymw ,

yes, Exception codes are valid. 
I tried to pull latest master on my server, but it still shows old code.
I think all latest code is not merged to master yet.

Can you please merge on master and confirm ?
so I can download that latest source and also do testing with test site.